stan/math/prim/mat/functor/map_rect.hpp

```diff
@@ -1,6 +1,7 @@
 #ifndef STAN_MATH_PRIM_MAT_FUNCTOR_MAP_RECT_HPP
 #define STAN_MATH_PRIM_MAT_FUNCTOR_MAP_RECT_HPP
 
+#include <stan/math/prim/scal/meta/return_type.hpp>
```

Comment:
I am not sure where to move given your comments. I order these things by dependency scal/arr/mat and then alphabetically.

Reply:
In `prim`, these should be strictly alphabetical. I know our codebase hasn't caught up yet, but that's what it should be. For `rev`, you'll want to bring in the `core.hpp` first, then everything else in alphabetical order (irrespective of scal/arr/mat).

The files that shouldn't follow this: the big includes like `stan/math/prim/mat.hpp`. These aren't because they need to include traits in the right order.